src/Language/PureScript/Parser/Declarations.hs

```diff
@@ -332,11 +332,13 @@ parseIdentifierAndValue :: TokenParser (String, Expr)
 parseIdentifierAndValue =
   do
     name <- C.indented *> lname
-    b <- P.option (Var $ Qualified Nothing (Ident name)) rest
+    b <- P.option (Var $ Qualified Nothing (Ident name)) (rest name)
     return (name, b)
-  <|> (,) <$> (C.indented *> stringLiteral) <*> rest
+  <|> (do name <- C.indented *> stringLiteral; b <- rest name; return (name, b))
   where
-  rest = C.indented *> colon *> C.indented *> parseValue
+  rest name = C.indented *> separator name *> C.indented *> parseValue
+  separator name = colon <|> P.try (P.lookAhead equals) *> fail (updateSyntaxError name) P.<?> ":"
```

Comment:
I would typically address this by either:

- Making the two constructs use the same syntax (i.e. updates and literals would share the opening `{` and label) and handling errors later, or
- Just using `<?>` to provide a better error for `:`. As I see it, the existing error is fine, but could be improved by giving `:` a name.

Anyway, let me think about this a bit. The reason I left this in Discussion was that I wanted to think about how to handle the remaining ambiguities in the parser. This is a fine fix, but I think there might be a different fix which will handle ambiguities better in general.